### PR TITLE
fix: dubious ownership when running `git` 

### PR DESCRIPTION
**Pull request by xingyaoww**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

Require #3271 to merge first.

`fatal: detected dubious ownership in repository at '/workspace'` if `git` related commands are executed.

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

- Fix workspace permission to fix dubious ownership
- Add test case involving `git` operations

---
**Other references**

### CODE REVIEW

**Comment by xingyaoww** [comment]
Porbbaly the last PR before I think we are ready to release 0.9 :D

**Review by neubig — APPROVED** [review]
LGTM, thanks!

**Comment by neubig** [comment]
But it seems that there are linter errors.

**Comment by xingyaoww** [comment]
@neubig just fixed it! Looks like duplicated function definition caused by merge

File: opendevin/runtime/client/client.py [hunk 1]
```diff
@@ -156,6 +170,20 @@ def _init_bash_shell(self, work_dir: str, username: str) -> None:
...
+            action = CmdRunAction(command=command)
+            action.timeout = 300
+            logger.debug(f'Executing init command: {command}')
+            obs: CmdOutputObservation = await self.run(action)
+            logger.debug(
+                f'Init command outputs (exit code: {obs.exit_code}): {obs.content}'
+            )
+            assert obs.exit_code == 0
+
+    logger.info('Bash init commands completed')
```
Comment (li-boxuan) [inline opendevin/runtime/client/client.py:185]: Wrong indentation?
```suggestion
        logger.info('Bash init commands completed')
```

Comment (xingyaoww) [reply opendevin/runtime/client/client.py]: good catch! thanks!

**Review by li-boxuan — APPROVED** [review]

**Pull request merged**